### DISCUSSION TITLE
How to make a self-hosted app built using Spin shareable?

### DISCUSSION THREAD

**Discussion by akulmehta** [serversideup/spin#129] [Q&A]
Let's say I have built an application which uses Spin (Laravel) and I would like to make it open source, such that anyone can clone the repository and self-host that application. What files should I add to the gitignore to not commit to the repository so that they are not copying any of my infrastructure files?

My understanding (and I am really naive about these things) is to just add `.infrastructure` to the gitignore file - but is there any other files which should also be part of the gitignore (apart from the obvious like `.env` and `vendor`)?

Also, in such a case, if a user clones this github repository, they would first need to install Spin and after that would they be able to do just `spin up`? Or would there need to be other steps they would have to do?

Guides/guidance for these would be useful and appreciated - Thanks.

**Comment by jaydrogers** [comment]
I'd love to explore distributing apps with Spin more too. I plan to circle back to [Bugflow](https://bugflow.io/) and use Spin to distribute it.

In the eyes of the docker image, the protections we have in place are using `.dockerignore` files: https://shisho.dev/blog/posts/how-to-use-dockerignore/

I actually have this set up to do this automatically: https://github.com/serversideup/spin/blob/c0693ddbf57db665de90b3ee0aff6e727875166f/lib/actions/init.sh#L44-L53

In regards to the `.gitignore`, we do that for you automatically too https://github.com/serversideup/spin/blob/c0693ddbf57db665de90b3ee0aff6e727875166f/lib/actions/init.sh#L41-L43

**Reply by akulmehta** [reply]
So currently when I commited the spin new laravel project, it also commits the `.spin-inventory.ini` and `.spin.yml` files along with the `.infrastructure` directory. Should I remove those?

Sorry I am just not understanding how someone can just take my repository and host their own instance separate from mine.

**Reply by jaydrogers** [reply]
Ahh, I see your goal now.

What you would want to do is build a Docker image that will be able to be distributed.

You'd leave the Spin files in there for development, etc, but then in something like GitHub Actions, you'd construct another Docker image that will be distributed to your users.

See our Financial Freedom project for example: https://github.com/serversideup/financial-freedom

We have some work to do on that yet, but it's similar to what you're trying to do

**Reply by akulmehta** [reply]
Yes (I think 😂) - this is what I want to do. So any guidance on how to take a `spin new laravel` to that would be great to document.
